main.c

```diff
@@ -0,0 +1,41 @@
+#include <stdio.h>
+#include <math.h>
+
+int main() {
+    float a, b, c;
+    float discriminant, x1, x2;
+
+    // Ввод коэффициентов
+    printf("Введите коэффициенты a, b, c для квадратного уравнения ax^2 + bx + c = 0\n");
+    printf("a = ");
+    scanf("%f", &a);
+    printf("b = ");
+    scanf("%f", &b);
+    printf("c = ");
+    scanf("%f", &c);
+
+    // Проверка, чтобы a не было равно 0
+    if (a == 0) {
+        printf("Это не квадратное уравнение.\n");
+        return 1;
+    }
+
+    // Вычисление дискриминанта
+    discriminant = b * b - 4 * a * c;
```

Comment:
Неявное преобразование типа.

```suggestion
    discriminant = b * b - 4.0 * a * c;
```